web/src/containers/activity/CostAllocateFFP.test.js

```diff
@@ -1,34 +1,41 @@
 import { shallow } from 'enzyme';
-import sinon from 'sinon';
 import React from 'react';
 
 import {
   CostAllocateFFPRaw as CostAllocateFFP,
   makeMapStateToProps,
   mapDispatchToProps
 } from './CostAllocateFFP';
-import { updateActivity } from '../../actions/activities';
+import {
+  setCostAllocationFFPFundingSplit,
+  setCostAllocationFFPOtherFunding
+} from '../../actions/editActivity';
 
-describe('the CostAllocateFFP component', () => {
-  const sandbox = sinon.createSandbox();
+jest.spyOn(React, 'useContext');
+React.useContext.mockImplementation(() => ({
+  index: 'activity index'
+}));
 
+describe('the CostAllocateFFP component', () => {
   const state = {
-    activities: {
-      byKey: {
-        key: {
-          costAllocation: {
-            '1066': {
-              other: 10,
-              ffp: {
-                federal: 90,
-                state: 10
-              }
-            },
-            '1067': {
-              other: 0,
-              ffp: {
-                federal: 10,
-                state: 90
+    apd: {
+      data: {
+        activities: {
+          0: {
+            costAllocation: {
+              '1066': {
+                other: 10,
+                ffp: {
+                  federal: 90,
+                  state: 10
+                }
+              },
+              '1067': {
+                other: 0,
+                ffp: {
+                  federal: 10,
+                  state: 90
+                }
               }
             }
           }
```

Comment:
This needs to be an array rather than object; e.g.:

```js
activities: [{
  costAllocation: {
    ...
  }
}],
```

The reason the test fails is because `activities` is an object here. The `selectActivityByIndex` selector does some safety checking on the requested index to make sure it's in-bounds for the array.  It does that by making sure the index is `< activities.length`, but an object doesn't have a `.length` property, so it ends up checking `index < undefined`, which is always false. So then `selectActivityByIndex` will always return null because the safety check fails.

Reply:
![thank you](https://media.giphy.com/media/kiLq8SSDcewve/giphy.gif)